dev-python/coloredlogs/coloredlogs-10.0.ebuild

```diff
@@ -0,0 +1,41 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="Colored stream handler for the logging module"
+HOMEPAGE="
+	https://pypi.org/project/coloredlogs/
+	https://github.com/xolox/python-coloredlogs
+	https://coloredlogs.readthedocs.io/en/latest/"
+SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="dev-python/humanfriendly[${PYTHON_USEDEP}]"
+
+DEPEND="test? (
+		>=dev-python/capturer-2.4[${PYTHON_USEDEP}]
+		>=dev-python/coverage-4.2[${PYTHON_USEDEP}]
+		>=dev-python/mock-1.0.1[${PYTHON_USEDEP}]
+		>=dev-python/verboselogs-1.7[${PYTHON_USEDEP}] )"
+
+PATCHES="${FILESDIR}/${P}-skip-cli-test.patch"
+
+python_test() {
+	# Sandbox violations
+	sed \
+		-e 's:test_system_logging:_&:g' \
+		-e 's:test_syslog_shortcut_simple:_&:g' \
+		-e 's:test_syslog_shortcut_enhanced:_&:g' \
+		-i ${PN}/tests.py || die
+	esetup.py test
+}
```

Comment:
You already include a patch to fix tests - why not put these into that same patch as well then? 

Besides making modifications inside `python_test` is bad because it makes these exact commands everytime you "enter" the phase with a different python version, so this sed should actually fail after 3.6 is done and you begin with 3.7. Unless something updated and tests are run in python build_dir (whic is possible too). However, you can still reduce multiple sed calls by making this operation where it belongs, `src_prepare`.

I'd still suggest using the patch.

Reply:
I hadn't touched the tests yet, but it seems they depend on `pytest-cov` which is a deprecated dependency:
```
error: Command '['/usr/bin/python3.7', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/var/tmp/portage/dev-python/coloredlogs-10.0/temp/tmpz52072j1', '--quiet', 'pytest-cov>=2.3.1']' returned non-zero exit status 1.
```
Seeing as the tests don't work without pytest-cov, should I get rid of them?

Reply:
Fixed this, I worked around the issue with `--cov` by using `distutils_enable_tests pytest` instead of `distutils_enable_tests setup.py` and manually specifying which file to run.

All the packages here now have tests enabled and working, and all except capturer have `docs` (capturer's docs directory is empty)